File: LC1351.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        def bsearch(arr):
            l, h = 0, len(arr)

            while l < h:
                m = (l + h)//2

                if arr[m] < 0:
                    h = m
                else:
                    l = m + 1
            
            return len(arr) - l
        
        c = 0
        for i in grid:
            c += bsearch(i)
        
        return c

File: LC1351.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int c = 0, n = grid[0].size();
        
        for(auto row : grid){
            int h = upper_bound(row.begin(), row.end(), 0, greater<int>()) - row.begin();
            c += n - h;
        }

        return c;
    }
};
